sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala

```diff
@@ -179,6 +182,22 @@ abstract class QueryPlan[PlanType <: QueryPlan[PlanType]] extends TreeNode[PlanT
 
   override def verboseString(maxFields: Int): String = simpleString(maxFields)
 
+  override def simpleStringWithNodeId(): String = {
+    val tag = new TreeNodeTag[Int]("operatorId")
```

Comment:
we can define it in `object QueryPlan`, no need to create it every time.

Reply:
@cloud-fan Thanks.. will do.